Add CashScript documentation

rkalis opened the pull request:
- Add pages for cashc, cashscript sdk, cashscript language, and getting started

I use inline code for the subheaders for function documentation. I need these subheaders to be seen as inline code for the structure of the document. But when they are longer than 25 characters they are converted from inline code to a code block.

I found this code:

```js
// Short use inline custom component, long use codeblock
const CodePreSplitter = ({ children }: BasicProps) => {
  if (children && children[0].length > 25) {
    return <Code fontSize={14}>{children}</Code>
  }
  return <Pre>{children}</Pre>
}
```

Would there be a way to keep my inline code as inline code regardless of the length?

SpicyPete commented:
Please update the PR target to `stage` instead of master.  
Looking into the CodeSplitter and wiring the pages up

rkalis commented:
I changed the CodeSplitter to look for newlines as inline code will never have a newline, while code blocks will always have a newline (at least at the end). This allows for accurate splitting between inline code and code blocks.

I also added `font-style: inherit` and `font-weight: inherit` to inline code, so bold / italic text gets preserved inside inline code. Example: ``**`contract.function(parameter)`**``.

rkalis commented:
I moved the CodeSplitter changes to the new storybook repository. The wiring stuff is included in the new bitcoincom-developer repository, but I think it is fine to merge it in here as well, as this can be addressed when the repo is cleaned up at a later point.
